apps/studio/components/interfaces/Storage/StoragePolicies/StoragePoliciesBucketRow.tsx

```diff
@@ -40,11 +40,13 @@ export const StoragePoliciesBucketRow = ({
 }: StoragePoliciesBucketRowProps) => {
   return (
     <Card>
-      <CardHeader className="flex flex-row w-full items-center justify-between gap-0 space-y-0">
-        <div className="flex items-center gap-3">
+      <CardHeader className="flex flex-row w-full items-center justify-between gap-2 space-y-0">
+        <div className="flex flex-1 min-w-0 items-center gap-3">
           <BucketIcon className="text-foreground-muted" size={16} strokeWidth={1.5} />
-          <CardTitle>{label}</CardTitle>
-          {bucket?.public && <Badge variant="warning">Public</Badge>}
+          <div className="flex flex-1 min-w-0 items-center gap-1.5">
+            <CardTitle className="truncate">{label}</CardTitle>
+            {bucket?.public && <Badge variant="warning">Public</Badge>}
+          </div>
```

Comment:
Fixes weird wrapping whilst preserving full bucket name (in most cases, before truncating).